packages/dds/pact-map/src/interfaces.ts

```diff
@@ -12,7 +12,28 @@ export interface IPactMapEvents extends ISharedObjectEvents {
 	/**
 	 * Notifies when a new value goes pending or has been accepted.
 	 */
-	(event: "pending" | "accepted", listener: (key: string) => void);
+	(event: "pending" | "accepted", listener: (key: string, sequenceNumber: number) => void);
+}
+
+/**
+ * Details of the accepted pact.
+ */
+export interface IAcceptedPactDetails<T> {
+	/**
+	 * The accepted value of the given type or undefined (typically in case of delete).
+	 */
+	value: T;
+
+	/**
+	 * The sequence number when the value was accepted, which will normally coincide with one of three possibilities:
+	 * - The sequence number of the "accept" op from the final client we expected signoff from
+	 * - The sequence number of the ClientLeave of the final client we expected signoff from
+	 * - The sequence number of the "set" op, if there were no expected signoffs (i.e. only the submitting client
+	 * was connected when the op was sequenced)
```

Comment:
I mostly-removed this optimization a while back (AB#5061 tracks the bit I missed).  Now we expect an accept from the submitting client too except for in the detached state (in which case the sequenceNumber will be 0 as you mention below).